Automattic/jetpack
# Masterbar: Register admin_color as user meta

What's better than writing an API endpoint? Not writing an API endpoint.

We can use Core's `/users` endpoint to retrieve and update the `admin_color` user meta, as long as we register it to make it available in the REST API.




Fixes #17763 

#### Changes proposed in this Pull Request:

* Added `class-` prefix to admin-color-scheme class to appease phpcs.
* Registers `admin_color` as a user meta field and makes it available in REST API requests.
* Adds unit tests, covering schema requests, retrieving the color scheme and updating the color scheme.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






Using [the REST API console](https://wordpress.org/plugins/rest-api-console/):
* In wp-admin, navigate to Tools > REST API console
* Run `GET /wp/v2/users/{id}/?_fields=meta` in the console and make sure the response contains `admin_color`.

##### GET:
![image](https://user-images.githubusercontent.com/1398304/98873356-18045f00-242d-11eb-8871-ccf66c219316.png)

##### POST
![image](https://user-images.githubusercontent.com/1398304/98875145-7aab2a00-2430-11eb-9c04-8096d93027b1.png)


To run unit tests:
* Follow [setup guidelines](https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md#unit-testing) if you don't have unit tests set up yet.
* Run `phpunit --testsuite masterbar`.




#### Proposed changelog entry for your changes:

* Adds `admin_color` value to meta information for the users REST API endpoint.
